AppWidget/glance-widget/src/main/java/com/example/android/glancewidget/TodoListGlanceWidget.kt

```diff
@@ -0,0 +1,152 @@
+import androidx.glance.text.FontWeight
+import androidx.glance.text.Text
+import androidx.glance.text.TextStyle
+
+private val groceryStringIds = listOf(
+    R.string.grocery_list_milk,
+    R.string.grocery_list_eggs,
+    R.string.grocery_list_tomatoes,
+    R.string.grocery_list_bacon,
+    R.string.grocery_list_butter,
+    R.string.grocery_list_cheese,
+    R.string.grocery_list_potatoes,
+    R.string.grocery_list_broccoli,
+    R.string.grocery_list_salmon,
+    R.string.grocery_list_yogurt
+)
+
+/**
+ * Glance widget that showcases how to use:
+ * - Compound buttons
+ * - LazyColumn
+ * - State management using GlanceStateDefinition
+ */
+class TodoListGlanceWidget : GlanceAppWidget() {
+
+    override var stateDefinition: GlanceStateDefinition<*> = PreferencesGlanceStateDefinition
+
+    @Composable
+    override fun Content() {
+        Column(
+            modifier = GlanceModifier
+                .fillMaxSize()
+                .background(day = Color.White, night = Color.DarkGray)
+                .appWidgetBackground()
+                .cornerRadius(16.dp)
+                .padding(8.dp)
+        ) {
+            val context = LocalContext.current
+            val prefs = currentState<Preferences>()
+            Text(
+                text = context.getString(R.string.glance_todo_list),
+                modifier = GlanceModifier
+                    .fillMaxWidth()
+                    .padding(8.dp),
+                style = TextStyle(fontWeight = FontWeight.Bold, fontSize = 18.sp),
+            )
+            CountChecked()
+            LazyColumn {
+                items(groceryStringIds) {
+                    val idString = it.toString()
+                    val checked = prefs[booleanPreferencesKey(idString)] ?: false
+                    CheckBox(
+                        text = context.getString(it),
+                        checked = checked,
+                        onCheckedChange = actionRunCallback<CheckboxClickAction>(
+                            actionParametersOf(
+                                toggledKey to idString,
+                            )
+                        ),
+                        modifier = GlanceModifier.padding(12.dp)
+                    )
+                }
+            }
+        }
+    }
+}
+
+private val toggledKey = ActionParameters.Key<String>("ToggledKey")
+
+class CheckboxClickAction : ActionCallback {
+    override suspend fun onRun(context: Context, glanceId: GlanceId, parameters: ActionParameters) {
+        val toggledStringId = requireNotNull(parameters[toggledKey]) {
+            "Add $toggledKey parameter in the ActionParameters."
+        }
+
+        val checked = requireNotNull(parameters[ToggleableStateKey]) {
+            "This action should only be called in response to toggleable events"
+        }
+        updateAppWidgetState(context, PreferencesGlanceStateDefinition, glanceId) {
+            it.toMutablePreferences()
```

Comment:
better name this variable. This part is a bit confusing

Reply:
Done